### ISSUE TITLE
}bedrock.hier.unwind do not unwind all with pRecursive = 1

### ISSUE THREAD

**Issue by cw-cleung** [cubewise-code/bedrock#62]
}bedrock.hier.unwind behave differently as Bedrock 3 unwind. 
For example with this hierarchy, 
![image](https://user-images.githubusercontent.com/37115385/61200711-f437e880-a714-11e9-8801-9e307a0fc5d2.png)
and this parameter 
![image](https://user-images.githubusercontent.com/37115385/61200752-0f0a5d00-a715-11e9-9a34-3e5cd8f889cc.png)
gives 
![image](https://user-images.githubusercontent.com/37115385/61200761-1af61f00-a715-11e9-9996-b5f42d559ee6.png)
Element "A", "B", "C" are still attached the "All".

In Bedrock 3 it should flat out all hierarchy with pRecursive = 1, including the direct children (A,B,C) under "All", gives 2 empty consol (A, B), and an orphan C.

**Comment by lotsaram** [comment]
@cw-cleung there was a recent fix to unwind. Are you using the latest version in your test?

**Comment by cw-cleung** [comment]
@lotsaram I'm using the one last modified on 20190524, same as the one in the main folder. Is there a newer one? 
![image](https://user-images.githubusercontent.com/37115385/61201488-3eba6480-a717-11e9-8e3d-6e1219077612.png)

**Comment by lotsaram** [comment]
Looks like the right version. Do you have a fix developed already?

**Comment by cw-cleung** [comment]
Not yet. my current workaround is to run it twice, first time with pRecursive = 1 and the second time with pRecursive = 0. 

I think the issue is this block of code (in prolog line 162) is not run when pRecursive = 1 in bedrock 4. But I'm not sure about the whole design so can't say for sure this is wrong. 

        nChildren = ElementComponentCount( pDim, pHier, pConsol );
        While( nChildren > 0 );
            # Unwind direct children of target hierarchy only
            sChild = ElementComponent( pDim, pHier, pConsol, nChildren );
            HierarchyElementComponentDelete( pDim, pHier, pConsol, sChild );
            If( pLogOutput = 1 );
                LogOutput( 'INFO', Expand( 'Deleting component "%sChild%" from consolidation %pConsol% in hierarchy "%pHier%" of "%pDim%" dimension.' ) );
            EndIf;
            nChildren = nChildren - 1;
        End;

**Comment by lotsaram** [comment]
Hi @cw-cleung 
The bug is actually in the setting of the "Bedrock.Descendant" attribute

```
If( ElementIsAncestor( pDim, pHier, pConsol, sElement ) = 1 );
    ElementAttrPutS( cAttrVal, pDim, pHier, sElement, cHierAttr );
 EndIf;
```

This will not set the attribute for the target consolidation itself (so the direct children will be skipped.
It just needs to be changed to the following

```
If( ElementIsAncestor( pDim, pHier, pConsol, sElement ) = 1 % pConsol @= sElement );
    ElementAttrPutS( cAttrVal, pDim, pHier, sElement, cHierAttr );
 EndIf;
```

**Comment by lotsaram** [comment]
implemented in #71
